tests/utils/test_autologging_utils.py

```diff
@@ -0,0 +1,107 @@
+import mlflow
+import pytest
+from mlflow.utils.autologging_utils import get_unspecified_default_args, \
+    log_fn_args_as_params
+
+
+# Example function signature we are testing on
+# def fn(arg1, default1=1, default2=2):
+#     pass
+
+
+two_default_test_args = [(['arg1', 'default1'], {'default2': 42},
+                         ['arg1', 'default1', 'default2'], [1, 2], {}),
+                         (['arg1', 'default1', 'default2'], {},
+                         ['arg1', 'default1', 'default2'], [1, 2], {}),
+                         (['arg1'], {'default1': 42, 'default2': 42},
+                         ['arg1', 'default1', 'default2'], [1, 2], {}),
+                         ([], {'arg1': 42, 'default1': 42, 'default2': 42},
+                         ['arg1', 'default1', 'default2'], [1, 2], {}),
+                         (['user_arg'], {'default1': 42},
+                         ['arg1', 'default1', 'default2'], [1, 2], {'default2': 2}),
+                         (['user_arg'], {'default2': 42},
+                         ['arg1', 'default1', 'default2'], [1, 2], {'default1': 1}),
+                         ([], {'arg1': 42, 'default1': 42},
+                         ['arg1', 'default1', 'default2'], [1, 2], {'default2': 2}),
+                         (['arg1', 'default1'], {},
+                         ['arg1', 'default1', 'default2'], [1, 2], {'default2': 2}),
+                         (['arg1'], {},
+                         ['arg1', 'default1', 'default2'], [1, 2], {'default1': 1, 'default2': 2}),
+                         ([], {'arg1': 42},
+                         ['arg1', 'default1', 'default2'], [1, 2], {'default1': 1, 'default2': 2})]
+
+
+@pytest.mark.large
```

Comment:
We could probably keep these tests small right? Are they slow to run?

Reply:
We could reduce them, but all the tests together only take about 1 second to run entirely.